addons/l10n_ro_edi/tests/test_xml_ubl_ro.py

```diff
@@ -0,0 +1,147 @@
+from odoo import Command
+from odoo.addons.l10n_account_edi_ubl_cii_tests.tests.common import TestUBLCommon
+from odoo.addons.account.tests.test_account_move_send import TestAccountMoveSendCommon
+from odoo.exceptions import UserError
+from odoo.tests import tagged
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestUBLRO(TestUBLCommon, TestAccountMoveSendCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref="ro"):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+        cls.company_data['company'].write({
+            'country_id': cls.env.ref('base.ro').id,
+            'state_id': cls.env.ref('base.RO_B').id,
+            'currency_id': cls.env.ref('base.RON').id,
+            'name': 'Hudson Construction',
+            'city': 'SECTOR1',
+            'zip': '010101',
+            'vat': 'RO1234567897',
+            'phone': '+40 123 456 789',
+            'street': "Strada Kunst, 3",
+            'invoice_is_ubl_cii': True,
+        })
+
+        cls.env['res.partner.bank'].create({
+            'acc_type': 'iban',
+            'partner_id': cls.company_data['company'].partner_id.id,
+            'acc_number': 'RO98RNCB1234567890123456',
+        })
+
+        cls.partner_a = cls.env['res.partner'].create({
+            'country_id': cls.env.ref('base.ro').id,
+            'state_id': cls.env.ref('base.RO_B').id,
+            'name': 'Roasted Romanian Roller',
+            'city': 'SECTOR3',
+            'zip': '010101',
+            'vat': 'RO1234567897',
+            'phone': '+40 123 456 780',
+            'street': "Rolling Roast, 88",
+            'ref': 'ref_partner_a',
+        })
+
+        cls.tax_19 = cls.env['account.tax'].create({
+            'name': 'tax_19',
+            'amount_type': 'percent',
+            'amount': 19,
+            'type_tax_use': 'sale',
+            'country_id': cls.env.ref('base.ro').id,
+        })
+
+    ####################################################
+    # Test export - import
+    ####################################################
+
+    def create_move(self, move_type):
```

Comment:
use `_generate_move` instead?

Reply:
@smetl
I think that would result in more code since I still need to add a few lines for the parameters. I prefer the current way as it's simpler :)
Or do you mean using _generate_move inside the create_move?

Reply:
```py
def create_move(self, move_type):
    return self._generate_move(
        self.env.company.partner_id,
        self.partner_a,
        send=True,
        move_type=move_type,
        invoice_line_ids=[
            Command.create({
                'product_id': self.product_a.id,
                'price_unit': 500.0,
                'tax_ids': [Command.set(self.tax_19.ids)],
            }),
            Command.create({
                'product_id': self.product_b.id,
                'price_unit': 1000.0,
                'tax_ids': [Command.set(self.tax_19.ids)],
            }),
        ],
    )
```